### PR TITLE
Refactoring of the makefile

### PR DESCRIPTION
**Pull request by tboerger**
I have refactored the makefile like I'm already using for other projects. It's prepared for cross-compiling Gitea for various platforms, you can see a list of the resulting binaries below. So what's next? Next steps will be the integration of `public` and `templates` into the binary and enabling Drone CI instead of Travis to directly provide publishing functionalities.

I have alredy opened this pull request to gather early feedback on my proposed changes.

* gitea-master-darwin-10.6-386
* gitea-master-darwin-10.6-amd64
* gitea-master-linux-386
* gitea-master-linux-amd64
* gitea-master-linux-arm-5
* gitea-master-linux-arm-6
* gitea-master-linux-arm-7
* gitea-master-linux-arm64
* gitea-master-linux-mips64
* gitea-master-linux-mips64le
* gitea-master-windows-4.0-386.exe
* gitea-master-windows-4.0-amd64.exe

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,74 +1,132 @@
...
+modules/bindata/bindata.go: $(BINDATA)
+	go-bindata -o=$@ -ignore="\\.go|README.md|TRANSLATORS" -pkg=bindata conf/...
+	go fmt $@
+
+.PHONY: javascripts
+javascripts: public/js/gogs.js
 
-fixme:
-	grep -rnw "FIXME" routers models modules
+.IGNORE: public/js/gogs.js
```
Comment (metalmatze) [inline Makefile]: This is gogs.js for now, until someone updates the frontend? Right?

**Comment by metalmatze** [comment]
Really solid `Makefile`. Just like all your `Makefile`s. 🎈

**Comment by metalmatze** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (tboerger) [reply Makefile]: Yes

**Comment by xinity** [comment]
Wow !!!!!!! simply awesome @tboerger

**Comment by xinity** [comment]
LGTM

**Comment by tboerger** [comment]
Rebased on latest master

File: .gitignore [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
+*.exe
+*.test
+*.prof
+
+coverage.out
+gitea
+
 *.db
 *.log
-log/
```
Comment (bkcsoft) [inline .gitignore:4]: Why skip this? (though it should be `/log` if anything ^.^)

**Comment by strk** [comment]
LGTM

File: .gitignore (hunk 2 above)
Comment (tboerger) [reply .gitignore:4]: Fixed

**Comment by tboerger** [comment]
I will create a followup PR for templates and public bindata and than also for Drone.

**Pull request merged**